addons/mail/static/src/models/thread_cache/thread_cache.js

```diff
@@ -203,8 +203,8 @@ function factory(dependencies) {
             if (!wasCacheRefreshRequested && this.threadViews.length === 0) {
                 // don't load message that won't be used
                 return false;
-                }
-            if (!wasCacheRefreshRequested && (this.isLoaded || this.isLoading)) {
+            }
+            if ((!wasCacheRefreshRequested && this.isLoaded) || this.isLoading) {
```

Comment:
Shouldn't it just be `||` between all parts of this condition?

Reply:
IMO, if the thread cache is loaded then we need to do the message fetch if a cache refresh is requested.
Otherwise if this is already loading then we don't need to do the same request again.
That's how I understand this

Reply:
Ok what you just said sounds correct indeed, so the only difference is if we have a request but it is already loading, we ignore the request. I wonder if it can lead to issues or not, but I can't think of any so I guess that's fine.

Reply:
Split the condition then, preventive return if loading on its own line.

Reply:
As it's the same thread cache you don't have any domain change so normally the current request is exactly the same so it seems natural to cancel the new one. To be sure we could cancel the old one and make the new but I think this will affect performances and not change anything at the end